module/zfs/zio.c

```diff
@@ -3943,11 +3961,14 @@ zio_vdev_io_assess(zio_t *zio)
 	/*
 	 * If we can't write to an interior vdev (mirror or RAID-Z),
 	 * set vdev_cant_write so that we stop trying to allocate from it.
+	 * Note, a distributed spare for a dRAID can potentially get an ENXIO
+	 * from one of its "children", but we don't want to mark the entire
+	 * drive as vdev_cant_write.
```

Comment:
Is "drive" talking about the distributed spare vdev, or the physical disk (concrete leaf) here?

Reply:
It's talking about a distributed spare but upon closer inspection this looks wrong.  It's setting `vdev_cant_write` for interior vdev's and a distributed spare has `->vdev_op_leaf` set so it doesn't apply.  I've gone ahead and removed this hunk.